dags/sources/tasks/airflow_logic/config_management.py

```diff
@@ -55,12 +56,35 @@ class DAGConfig(BaseModel):
     ]
     combine_columns_categories: list[str] = []
     dechet_mapping: dict = {}
-    endpoint: HttpUrl
+    s3_connection_id: str | None = None
+    endpoint: str  # HttpURL or s3URL
     label_bonus_reparation: Optional[str] = None
     product_mapping: dict
     source_code: Optional[str] = None
     validate_address_with_ban: bool = False
 
+    @field_validator("endpoint")
+    def validate_endpoint(cls, v):
```

Comment:
```suggestion
    def validate_endpoint(cls, value):
```

Je suis pas un grand fan des variables a une lettre, sauf si c'est une convention acceptée dans l'écosystème qu'on utilise.
Je ne sais pas si c'est le cas ici

Reply:
je renomme `v` en `endpoint`